relay-server/src/actors/project_buffer.rs

```diff
@@ -189,11 +213,130 @@ impl Service for BufferService {
+            .filename(PathBuf::from("sqlite://").join(path))
+            .journal_mode(SqliteJournalMode::Wal)
+            .create_if_missing(true);
+
+        let db = SqlitePoolOptions::new().connect_with(options).await?;
+        Ok(Self { db })
+    }
+
+    /// Handles the enqueueing messages into the internal buffer.
+    async fn handle_enqueue(&self, message: Enqueue) -> Result<(), BufferError> {
+        let Enqueue {
+            key,
+            value: mut managed_envelope,
+        } = message;
+
+        managed_envelope.spool();
+        let envelope_bytes = managed_envelope.take_envelope().to_vec()?;
+        sqlx::query("INSERT INTO envelopes (own_key, sampling_key, envelope) VALUES (?, ?, ?)")
+            .bind(key.own_key.to_string())
+            .bind(key.sampling_key.to_string())
+            .bind(envelope_bytes)
+            .execute(&self.db)
+            .await?;
+        Ok(())
+    }
+
+    /// Handles the dequeueing messages from the internal buffer.
+    ///
+    /// This method removes the envelopes from the buffer and stream them to the sender.
+    async fn handle_dequeue(&self, message: DequeueMany) -> Result<(), BufferError> {
+        let DequeueMany { keys, sender } = message;
+
+        for key in keys {
+            let mut envelopes = sqlx::query(
+                "DELETE FROM envelopes WHERE own_key = ? AND sampling_key = ? RETURNING envelope",
+            )
+            .bind(key.own_key.to_string())
+            .bind(key.sampling_key.to_string())
+            .fetch(&self.db);
+
+            while let Some(row) = envelopes.try_next().await? {
+                let envelope_bytes_slice: &[u8] = row.try_get("envelope")?;
+                let envelope_bytes = bytes::Bytes::from(envelope_bytes_slice);
+                let envelope = Envelope::parse_bytes(envelope_bytes)?;
+
+                // TODO: do not use the "standalone"  here , rather make sure to hande over the
+                // permit. We have to plug into the system.
+                let managed_envelope = ManagedEnvelope::standalone(envelope);
+                sender.send(managed_envelope).ok();
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Handles the remove request.
+    ///
+    /// This remove all the envelopes from the internal buffer for the provided keys.
+    /// If any of the provided keys are still have the envelopes, the error will be logged with the
+    /// number of envelopes dropped for the specific project key.
+    async fn handle_remove(&self, message: RemoveMany) -> Result<(), BufferError> {
+        let RemoveMany { project_key, keys } = message;
+
+        let mut count = 0;
+        for key in keys {
+            let result =
+                sqlx::query("DELETE FROM envelopes where own_key = ? AND sampling_key = ?")
+                    .bind(key.own_key.to_string())
+                    .bind(key.sampling_key.to_string())
+                    .execute(&self.db)
+                    .await?;
```

Comment:
What is the use case for dropping envelopes from disk?

Reply:
Right now it mimics the current behaviour, if the project expired (maybe it was deleted etc.) we will drop everything we have for it. This normally should not happen. For example, the memory implementation does not drop anything, otherwise we would get many errors.